lib/db_utils.go

```diff
@@ -304,14 +305,23 @@ var (
 	//   SellingDAOCoinCReatorPKID [33]byte
 	//   ScaledPrice [256]byte
 	//   BlockHeight [32]byte
-	// > -> <DAOCoinLimitOrder>
+	//   OrderID [32]byte
+	// > -> <DAOCoinLimitOrderEntry>
+	//
+	// This index allows users to query for a single order by ID.
+	// This is useful in e.g. cancelling an order.
+	// <
+	//   _PrefixDAOCoinLimitOrderByOrderID
+	//   OrderID [32]byte
+	// > -> <DAOCoinLimitOrderEntry>
```

Comment:
Interesting. I was thinking you wouldn't need the entries here anymore, and could just "normalize" it to make it so you just look it up from the OrderID index when you need it. But actually I like what you did here because it keeps the lookups fast and requires minimal code changes. Just make sure the code that updates these three indexes always keeps things 100% consistent. That means updating all 3 indexes whenever you modify a single order.